test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -294,6 +300,67 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
                 sendMalformedRequest: true);
         }
 
+        [Fact]
+        public async Task OnCompletedExceptionShouldNotPreventAResponse()
+        {
+            var hostBuilder = TransportSelector.GetWebHostBuilder()
+                .UseKestrel()
+                .UseUrls("http://127.0.0.1:0/")
+                .Configure(app =>
+                {
+                    app.Run(async context =>
+                    {
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+                        context.Response.OnCompleted(async () => throw new Exception());
```

Comment:
Nit: I think you can get rid of the pragmas by simply removing `async` from your callbacks.

Reply:
OnCompleted takes a Task as it's parameter so I can change this to `Task.Run(() => throw Exception())`.

Reply:
Oh right I can use the overload, silly me.